src/java.base/share/native/libjli/java.c

```diff
@@ -946,11 +951,15 @@ AddOption(char *str, void *info)
      * 'default' sizes (either from JVM or system configuration, e.g. 'ulimit -s' on linux),
      * and is not itself a small stack size that will be rejected. So we ignore -Xss0 here.
      */
-    if (JLI_StrCCmp(str, "-Xss") == 0) {
+
+    /* In OpenJ9 -Xmso is used to set native stack size instead of -Xss. -Xss is used to
+     * set Java thread size only, which is handled in the JVM code
+     */
+    if (JLI_StrCCmp(str, "-Xmso") == 0) {
         jlong tmp;
-        if (parse_size(str + 4, &tmp)) {
+        if (parse_size(str + 5, &tmp)) {
             threadStackSize = tmp;
-            if (threadStackSize > 0 && threadStackSize < (jlong)STACK_SIZE_MINIMUM) {
+            if ((threadStackSize > 0) && (threadStackSize < (jlong)STACK_SIZE_MINIMUM)) {
```

Comment:
Normally we would add parentheses as you've done, but we'd prefer to keep changes relative to upstream as small as reasonably possible. Please revert the change to this line.